experimental/segmenter/src/complex.rs

```diff
@@ -83,12 +83,13 @@ pub fn complex_language_segment_utf16(
         }
 
         if let Some(payload) = dictionary.best(str_per_lang[0] as u32) {
-            if let Ok(segmenter) = DictionarySegmenter::try_new_unstable(payload) {
-                let breaks = segmenter.segment_utf16(&str_per_lang);
-                let mut r: Vec<usize> = breaks.map(|n| offset + n).collect();
-                result.append(&mut r);
-                offset += str_per_lang.len();
-                continue;
+            if let Some(grapheme) = grapheme {
+                if let Ok(segmenter) = DictionarySegmenter::try_new_unstable(payload, grapheme) {
+                    let breaks = segmenter.segment_utf16(&str_per_lang);
+                    result.extend::<Vec<usize>>(breaks.map(|n| offset + n).collect());
```

Comment:
`optional`: I don't think you need to collect here. `extend` accepts an iterator, so you should be able to just do:

```rust
result.extend(breaks.map(|n| offset + n));
```

playground: https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=0e406bf0d110c82491e1b04e58b4dbde